instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.java

```diff
@@ -0,0 +1,194 @@
+                span -> AwsSpanAssertions.sns(span, 2, "SNS.Publish", trace.getSpan(1)),
+                span ->
+                    AwsSpanAssertions.sqs(
+                        span,
+                        3,
+                        "SQS.ReceiveMessage",
+                        queueUrl,
+                        null,
+                        SpanKind.CONSUMER,
+                        trace.getSpan(2)),
+                span -> CamelSpanAssertions.sqsConsume(span, 4, queueName, trace.getSpan(2))),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.ReceiveMessage", queueUrl)),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.DeleteMessage", queueUrl)),
+        // camel polling
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.ReceiveMessage", queueUrl)));
+
+    try {
+      awsConnector.purgeQueue(queueUrl);
+    } catch (PurgeQueueInProgressException e) {
+      logger.warn("Throttled by AWS trying to purge queue, try doing it manually.");
+    }
+    camelApp.stop();
+  }
+
+  private static void waitAndClearSetupTraces(String queueUrl, String queueName) {
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.CreateQueue", queueUrl, queueName)),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.GetQueueAttributes", queueUrl)),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.SetQueueAttributes", queueUrl)),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sns(span, 0, "SNS.CreateTopic", null)),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sns(span, 0, "SNS.Subscribe", null)),
+        // test message
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.ReceiveMessage", queueUrl)));
+    testing.clearData();
+  }
+
+  Map<String, String> setupTestInfrastructure(String queueName, String topicName) {
+    // setup infra
+    String queueUrl = awsConnector.createQueue(queueName);
+    String queueArn = awsConnector.getQueueArn(queueUrl);
+    awsConnector.setQueuePublishingPolicy(queueUrl, queueArn);
+    String topicArn = awsConnector.createTopicAndSubscribeQueue(topicName, queueArn);
+
+    // consume test message from AWS
+    awsConnector.receiveMessage(queueUrl);
+
+    Map<String, String> metaData = new HashMap<>();
+    metaData.put("queueUrl", queueUrl);
+    metaData.put("topicArn", topicArn);
+    return metaData;
+  }
```

Comment:
Nit: instead of a map, consider adding a small `private static final` class that'll hold these two vars.